diode/client.go

```diff
@@ -61,21 +67,43 @@ var (
 	allowedSchemesRe = regexp.MustCompile(`grpc|grpcs|http|https`)
 )
 
-// loadCerts loads the system x509 cert pool
-func loadCerts() *x509.CertPool {
+// loadCerts loads the x509 cert pool from custom cert file or system certs
+func loadCerts(certFile string) (*x509.CertPool, error) {
+	if certFile != "" {
+		certData, err := os.ReadFile(certFile)
```

Comment:
The use case is reading a single certificate, correct? Would we benefit from specifying a directory of certificates to read from? I believe that would work with the `*x509.CertPool`.